### PR TITLE
fa2-contracts doc

### PR DESCRIPTION
**Pull request by emishur**

### CODE REVIEW

File: packages/fa2-contracts/README.md [hunk 1]
```diff
@@ -5,55 +5,434 @@ contracts.
...
+
+compiled contract to ~/my_project/dist/my_contract.tz file
+```
+
+### Generate TypeScript Code
+
+`type-script` command generates a TypeScript interface for the contract from the
+specification file. The command takes two arguments: name of the specification
+file and name of the resulting TypeScript file. The resulting file will be created
+in TypeScript source code location (`./src` in our case).
```
Comment (alex-kooper) [inline packages/fa2-contracts/README.md:266]: I think it should be "two arguments: **the** name of the specification file and **the** name of the resulting"

File: packages/fa2-contracts/README.md [hunk 2]
```diff
@@ -5,55 +5,434 @@ contracts.
...
+
+```sh
+$ yarn tzgen type-script my_contract.json my_contract.ts
+contract interface source code file ~/my_project/src/my_contract.ts is generated
+```
+
+The resulting file will contain two functions: `createStorage` to create a storage
+object for the contract origination and `createContractInterface` to get a strongly
+typed interface to interact with the contract on the block chain. To use and compile
+a generated file, your package must include dependencies on `@taquito/taquito` and
```
Comment (alex-kooper) [inline packages/fa2-contracts/README.md:278]: I think it should be "your package must have include dependencies on"

File: packages/fa2-contracts/README.md [hunk 3]
```diff
@@ -5,55 +5,434 @@ contracts.
...
+able mint and burn new tokens.
+
+* `withNoMinter()` contract cannot mint or burn tokens
+* `withMint()` contract can mint new tokens
+* `withBurn()` contract can burn tokens
+* `withFreeze()` mint/burn functionality can be frozen
+
+### Minter Admin Combinator
+
+If a contract has mint/burn functionality, you haver to select
```
Comment (alex-kooper) [inline packages/fa2-contracts/README.md:371]: Typo, it should be "have" not "haver"

File: packages/fa2-contracts/README.md [hunk 4]
```diff
@@ -5,55 +5,434 @@ contracts.
...
+[minter admin](#minter-admin) implementation as well.
+
+* `withNoMinterAdmin()` no minter admin. Anyone can mint/burn tokens
```
Comment (alex-kooper) [inline packages/fa2-contracts/README.md:374]: Probably a period "." at the end of the second sentence.

**Review by alex-kooper — CHANGES_REQUESTED** [review]

File: packages/fa2-contracts/README.md [hunk 5]
```diff
@@ -5,55 +5,434 @@ contracts.
...
+
+This feature defines access to mint and burn functionality defined by the
+[minter](#minter-functionality) feature.
+
+* `USE_NULL_MINTER_ADMIN` - contract does not have a minter admin. If either
+  `CAN_MINT` or `CAN_BURN` feature is selected, anyone can mint or burn tokens.
+  This is also the default option if no mint or burn feature are selected
+* `USE_ADMIN_AS_MINTER` - contract admin can also mint and burn tokens
+* `USE_MULTI_MINTER_ADMIN` - contract can have multiple minter admins that can
+  mint and burn tokens. Minter admin list is separate from the contract admin(s)
```
[the remark this answers is not in the document]
Comment (emishur) [reply packages/fa2-contracts/README.md:100]: contract admin(s) may be a list or may be not (single admin)

File: packages/fa2-contracts/README.md (hunk 2 above)
Comment (emishur) [reply packages/fa2-contracts/README.md:278]: I think it should be left as is

**Review by alex-kooper — APPROVED** [review]

**Pull request merged**